core/corehttp/gateway_indexPage.go

```diff
@@ -0,0 +1,160 @@
+package corehttp
+
+import (
+	"html/template"
+	"path"
+)
+
+// structs for directory listing
+type listingTemplateData struct {
+	Listing  []directoryItem
+	Path     string
+	BackLink string
+}
+
+type directoryItem struct {
+	Size uint64
+	Name string
+	Path string
+}
+
+// Directory listing template
+var listingTemplate = template.Must(template.New("dir").Funcs(template.FuncMap{"iconFromExt": iconFromExt}).Parse(`
+<!DOCTYPE html>
+<html>
+		<head>
+				<meta charset="utf-8" />
+				<!-- TODO: seed these - maybe like the starter ex or the webui? -->
+				<link rel="stylesheet" href="/ipfs/QmXB7PLRWH6bCiwrGh2MrBBjNkLv3mY3JdYXCikYZSwLED/bootstrap.min.css"/>
+				<!-- helper to construct this is here: https://github.com/cryptix/exp/blob/master/imgesToCSSData/convert.go -->
```

Comment:
I have good experience with [jteeuwen/go-bindata](https://github.com/jteeuwen/go-bindata) and [shurcooL/vfsgen](https://github.com/shurcooL/vfsgen) to embedd files with `go-generate`.

I just wasn't sure if we want this for a single file. I'll fix the indenting.